src/auto.rs

```diff
@@ -74,17 +75,57 @@ where
     }
 }
 
+pub enum AutoReaderOptions {
+    Xls(XlsOptions),
+    Xlsx(XlsxOptions),
+    Xlsb(XlsbOptions),
+    Ods(OdsOptions),
+}
+
+impl ReaderOptions for AutoReaderOptions {
+    fn with_header_row(self, header_row: u32) -> Self {
+        match self {
+            AutoReaderOptions::Xls(e) => AutoReaderOptions::Xls(e.with_header_row(header_row)),
+            AutoReaderOptions::Xlsx(e) => AutoReaderOptions::Xlsx(e.with_header_row(header_row)),
+            AutoReaderOptions::Xlsb(e) => AutoReaderOptions::Xlsb(e.with_header_row(header_row)),
+            AutoReaderOptions::Ods(e) => AutoReaderOptions::Ods(e.with_header_row(header_row)),
+        }
+    }
+}
+
 impl<RS> Reader<RS> for Sheets<RS>
 where
     RS: std::io::Read + std::io::Seek,
 {
     type Error = Error;
+    type Options = AutoReaderOptions;
 
     /// Creates a new instance.
     fn new(_reader: RS) -> Result<Self, Self::Error> {
         Err(Error::Msg("Sheets must be created from a Path"))
     }
 
+    fn set_options(&mut self, options: Self::Options) {
+        match *self {
+            Sheets::Xls(ref mut e) => match options {
+                AutoReaderOptions::Xls(opts) => e.set_options(opts),
+                _ => unreachable!(),
+            },
+            Sheets::Xlsx(ref mut e) => match options {
+                AutoReaderOptions::Xlsx(opts) => e.set_options(opts),
+                _ => unreachable!(),
+            },
+            Sheets::Xlsb(ref mut e) => match options {
+                AutoReaderOptions::Xlsb(opts) => e.set_options(opts),
+                _ => unreachable!(),
+            },
+            Sheets::Ods(ref mut e) => match options {
+                AutoReaderOptions::Ods(opts) => e.set_options(opts),
+                _ => unreachable!(),
+            },
+        }
+    }
```

Comment:
```suggestion
    fn set_options(&mut self, options: Self::Options) {
        match (*self, options) {
            (Sheets::Xls(ref mut e), AutoReaderOptions::Xls(opts)) => e.set_options(opts),
            (Sheets::Xlsx(ref mut e), AutoReaderOptions::Xlsx(opts)) => e.set_options(opts),
            (Sheets::Xlsb(ref mut e), AutoReaderOptions::Xlsb(opts)) => e.set_options(opts),
            (Sheets::Ods(ref mut e), AutoReaderOptions::Ods(opts)) => e.set_options(opts),
            _ => unreachable!(),
        }
    }
```

Reply:
I first tried that but I had the `move occurs because `*self` has type `Sheets<RS>`, which does not implement the `Copy` trait`
But in fact we can just switch to maching on reference itself! And it should be faster :)

Reply:
[dd33174](https://github.com/tafia/calamine/pull/453/commits/dd33174899447bdb02d34b9eee622eeafd0eb9bd)